src/pages/index.js

```diff
@@ -66,57 +49,155 @@ const cardGallerySelector = ".gallery__grid";
 const section = new Section(
-  { items: initialCards, renderer: createCard },
+  { items: cardHolder, renderer: createCard },
   cardGallerySelector
 );
-section.renderItems();
+
+const handleDeleteConfirm = (currentCard, cardId) => {
+  const isNotSubmitted = !confirmModal.getIsSubmitted();
+  console.log(`not submitted: ${isNotSubmitted}`);
+  if (!confirmModal.getIsSubmitted()) {
+    console.log("no submission");
+  } else {
+    newApi.deleteCard({ cardId: cardId }).then((res) => {
+      console.log(res, cardId);
+    });
+    //remove card from page
+    currentCard.removeCard();
+    console.log(`deleting: ${cardId}`);
+  }
+};
+
+//"Are you sure?" confirmation form to delete
+const confirmModal = new ModalWithConfirm({
+  modalSelector: "#confirm-modal",
+  handleConfirmAction: handleDeleteConfirm,
+});
+confirmModal.setEventListeners();
+
+//the API's data (response) is the new "initial" array
+newApi.getInitialCards().then((data) => {
+  const section = new Section(
+    { items: data, renderer: createCard },
+    cardGallerySelector
+  );
+  section.renderItems();
+});
 
 //make global instance of Userinfo class
 const userInfo = new UserInfo({
   profileNameSelector: ".profile__title",
   profileJobSelector: ".profile__subtitle",
+  profilePictureSelector: ".profile__picture",
 });
 
 //function to save person. this takes data from the form,
 //then lets userinfo use it
-const handleProfileFormSubmit = (data) => {
+const handleProfileFormSubmit = (inputData) => {
+  console.log(inputData);
   //add the info to the page
-  userInfo.setUserInfo(data.title, data.subtitle);
+  newApi
+    .editProfile({ name: inputData.title, about: inputData.subtitle })
+    .then((data) => {
+      console.log(data);
+    });
+  userInfo.setUserInfo({ name: inputData.title, job: inputData.subtitle });
 };
 
 //function to create new card img
-const handleImgFormSubmit = (data) => {
+const handleImgFormSubmit = (inputData) => {
+  console.log(inputData);
   //create new card
-  const newCard = {
-    name: data.imgTitle,
-    link: data.imgURL,
-    alt: data.imgTitle,
-  };
-  place = "prepend";
-  createCard(newCard);
+  console.log(inputData.imgTitle, inputData.imgURL);
+  newApi
+    .addNewCard({ name: inputData.imgTitle, link: inputData.imgURL })
+    .then((data) => {
+      const newCard = {
+        name: data.name,
+        link: data.link,
+        alt: data.name,
+        _id: data._id,
+      };
+      place = "prepend";
+      createCard(newCard);
+    });
 };
 
+/*just try this valid link from practicum:
+https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/logo.svg
+ to enter into the input to test it for convenience.
+ */
+const handlePfpFormSubmit = (inputData) => {
+  console.log(inputData.pfpURL);
+  console.log(`change pfp url to: ${inputData.pfpURL} `);
+  newApi.changePfp({ avatar: `${inputData.pfpURL}` }).then((res) => {
+    console.log(res);
+  });
+  userInfo.setUserPicture({ picture: inputData.pfpURL });
```

Comment:
Needs correction: you should update the ui only after the successful api response in `then` block